six/modules/python/six.sicd/source/six_sicd.i

```diff
@@ -39,20 +39,83 @@ using namespace six;
 six::sicd::ComplexData * asComplexData(six::Data* data);
 
 /* Need this because we can't really do it on the python side of things */
-six::sicd::ComplexData * asComplexData(six::Data* data) 
+six::sicd::ComplexData * asComplexData(six::Data* data)
 {
   six::sicd::ComplexData * complexData = dynamic_cast<six::sicd::ComplexData*>(data);
   if( !complexData )
   {
     throw except::BadCastException();
   }
-  else 
+  else
   {
     return complexData;
   }
 }
 
+void writeNITF(std::string fileName, std::vector<std::string> schemaPaths,
```

Comment:
Pass in strings and vector of strings by const ref.  Also, use `pathname` rather than `fileName` throughout (we use `pathname` when it's the directory plus filename and `filename` when it doesn't contain the directory.

Pass `data` in by const ref rather than non-const pointer.